src/client/http_client.rs

```diff
@@ -664,23 +665,48 @@ impl QueryRequest {
                 scope,
                 timeout,
                 transaction,
+                parameters,
             } => {
-                let mut json = HashMap::new();
+                let mut json: HashMap<String, serde_json::Value> = HashMap::new();
                 if let Some(scope) = scope {
                     let ctx = format!("`default`:`{}`.`{}`", scope.0, scope.1);
-                    json.insert("query_context", ctx);
+                    json.insert("query_context".to_string(), serde_json::Value::String(ctx));
                 }
 
-                json.insert("statement", statement.to_string());
-                json.insert("timeout", timeout.to_string());
+                json.insert(
+                    "statement".to_string(),
+                    serde_json::Value::String(statement.to_string()),
+                );
+                json.insert(
+                    "timeout".to_string(),
+                    serde_json::Value::String(timeout.to_string()),
+                );
                 if let Some(txn) = transaction {
                     if let Some(t) = txn.tx_timeout.clone() {
-                        json.insert("txtimeout", format!("{}ms", t.as_millis()));
+                        json.insert(
+                            "txtimeout".to_string(),
+                            serde_json::Value::String(format!("{}ms", t.as_millis())),
+                        );
                     }
                     if let Some(id) = txn.tx_id.clone() {
-                        json.insert("txid", id);
+                        json.insert("txid".to_string(), serde_json::Value::String(id));
+                    }
+                }
+
+                if let Some(params) = parameters {
+                    match params {
+                        serde_json::Value::Array(_) => {
+                            json.insert("args".to_string(), params.clone());
+                        }
+                        serde_json::Value::Object(map) => {
+                            for (k, v) in map.iter() {
+                                json.insert(format!("${}", *k), v.clone());
```

Comment:
We're going to need to add a check here to see if the key already has a $ prefix, if so then don't add it ourselves.

Reply:
I also think we should just convert this to &str rather than .to_string() everywhere else.